From 9d52811e908031277402c31e53de2aee637227c4 Mon Sep 17 00:00:00 2001
From: Fredrik Mellbin <fredrik.mellbin@gmail.com>
Date: Mon, 12 Oct 2015 20:58:55 +0200
Subject: [PATCH] Fix imwri compilation issues

---
 src/filters/imwri/imwri.cpp | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/filters/imwri/imwri.cpp b/src/filters/imwri/imwri.cpp
index e0c97dd..8eb1ce5 100644
--- a/src/filters/imwri/imwri.cpp
+++ b/src/filters/imwri/imwri.cpp
@@ -33,9 +33,8 @@
 #include <algorithm>
 #include <memory>
 
-#if MAGICKCORE_QUANTUM_DEPTH != 16
-#error "Only Quantum Depth 16 ImageMagick builds are supported"
-#endif
+
+
 
 #ifdef _WIN32
 #define WIN32_LEAN_AND_MEAN
@@ -46,6 +45,8 @@
 #include <windows.h>
 #endif
 
+// Because ImageMagick has no idea how to make sane headers
+using namespace MagickCore;
 
 //////////////////////////////////////////
 // Shared
