diff --git a/include/VSHelper.h b/include/VSHelper.h
index e81cdb3..daf753b 100644
--- a/include/VSHelper.h
+++ b/include/VSHelper.h
@@ -46,9 +46,13 @@
 #define VSMAX(a,b) ((a) > (b) ? (a) : (b))
 #define VSMIN(a,b) ((a) > (b) ? (b) : (a))
 
-#ifdef __cplusplus
+#ifdef __cplusplus 
 /* A nicer templated malloc for all the C++ users out there */
+#if __cplusplus >= 201103L || defined(_MSC_VER)
 template<typename T=void>
+#else
+template<typename T>
+#endif
 static inline T* vs_aligned_malloc(size_t size, size_t alignment) {
 #ifdef _WIN32
     return (T*)_aligned_malloc(size, alignment);
diff --git a/include/VapourSynth.h b/include/VapourSynth.h
index 282b0f1..c2cddab 100644
--- a/include/VapourSynth.h
+++ b/include/VapourSynth.h
@@ -30,7 +30,11 @@
 /* Convenience for C++ users. */
 #ifdef __cplusplus
 #    define VS_EXTERN_C extern "C"
-#    define VS_NOEXCEPT noexcept
+#    if __cplusplus >= 201103L || defined(_MSC_VER)
+#        define VS_NOEXCEPT noexcept
+#    else
+#        define VS_NOEXCEPT
+#    endif
 #else
 #    define VS_EXTERN_C
 #    define VS_NOEXCEPT
