From d1e80c6a9d6efe7921300c01ffc0f311927ba443 Mon Sep 17 00:00:00 2001
From: dmo <marillat@users.noreply.github.com>
Date: Sun, 26 Sep 2021 22:25:42 +0200
Subject: [PATCH] Add meson build

---
 meson.build | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)
 create mode 100644 meson.build

diff --git a/meson.build b/meson.build
new file mode 100644
index 0000000..f741289
--- /dev/null
+++ b/meson.build
@@ -0,0 +1,21 @@
+project('ocr', 'c', 'cpp',
+  version : '1',
+  default_options : ['warning_level=3'])
+
+add_project_arguments('-ffast-math', language : 'c')
+
+sources = [
+     'src/ocr.c'
+]
+
+vapoursynth_dep = dependency('vapoursynth', version: '>=55').partial_dependency(compile_args : true, includes : true)
+libtesseract_dep = dependency('tesseract')
+
+deps = [vapoursynth_dep, libtesseract_dep]
+
+shared_module('ocr', sources,
+  dependencies : deps,
+  install : true,
+  install_dir : join_paths(vapoursynth_dep.get_pkgconfig_variable('libdir'), 'vapoursynth'),
+  gnu_symbol_visibility : 'hidden'
+)
