diff --git a/Makefile.am b/Makefile.am
index 5f65ec9..5c00af6 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -4,25 +4,27 @@ AM_CXXFLAGS = -std=c++14 $(common_cflags)
 
 AM_CPPFLAGS = $(VapourSynth_CFLAGS)
 
-lib_LTLIBRARIES = libeedi3.la
+lib_LTLIBRARIES = libeedi3m.la
+noinst_LTLIBRARIES =
+libeedi3m_la_LIBADD =
 
-libeedi3_la_SOURCES = EEDI3/EEDI3.cpp \
-                      EEDI3/EEDI3.hpp \
-                      EEDI3/shared.hpp \
-                      EEDI3/vectorclass/instrset.h \
-                      EEDI3/vectorclass/instrset_detect.cpp
+libeedi3m_la_SOURCES = EEDI3/EEDI3.cpp \
+                       EEDI3/EEDI3.hpp \
+                       EEDI3/shared.hpp \
+                       EEDI3/vectorclass/instrset.h \
+                       EEDI3/vectorclass/instrset_detect.cpp
 
 if VS_TARGET_CPU_X86
-libeedi3_la_SOURCES += EEDI3/EEDI3_SSE2.cpp \
-                       EEDI3/vectorclass/vectorclass.h \
-                       EEDI3/vectorclass/vectorf128.h \
-                       EEDI3/vectorclass/vectorf256.h \
-                       EEDI3/vectorclass/vectorf256e.h \
-                       EEDI3/vectorclass/vectori128.h \
-                       EEDI3/vectorclass/vectori256.h \
-                       EEDI3/vectorclass/vectori256e.h
+noinst_LTLIBRARIES += libsse4.la libavx2.la
 
-noinst_LTLIBRARIES = libsse4.la libavx2.la
+libeedi3m_la_SOURCES += EEDI3/EEDI3_SSE2.cpp \
+                        EEDI3/vectorclass/vectorclass.h \
+                        EEDI3/vectorclass/vectorf128.h \
+                        EEDI3/vectorclass/vectorf256.h \
+                        EEDI3/vectorclass/vectorf256e.h \
+                        EEDI3/vectorclass/vectori128.h \
+                        EEDI3/vectorclass/vectori256.h \
+                        EEDI3/vectorclass/vectori256e.h
 
 libsse4_la_SOURCES = EEDI3/EEDI3_SSE4.cpp
 libsse4_la_CXXFLAGS = $(AM_CXXFLAGS) -msse4.1
@@ -30,15 +32,15 @@ libsse4_la_CXXFLAGS = $(AM_CXXFLAGS) -msse4.1
 libavx2_la_SOURCES = EEDI3/EEDI3_AVX2.cpp
 libavx2_la_CXXFLAGS = $(AM_CXXFLAGS) -mfma -mavx2
 
-libeedi3_la_LIBADD = libsse4.la libavx2.la
+libeedi3m_la_LIBADD += libsse4.la libavx2.la
 endif
 
 if OPENCL
-libeedi3_la_SOURCES += EEDI3/EEDI3CL.cpp \
-                       EEDI3/EEDI3CL.hpp \
-                       EEDI3/EEDI3CL_SSE2.cpp
+noinst_LTLIBRARIES += libclsse4.la libclavx2.la
 
-noinst_LTLIBRARIES = libclsse4.la libclavx2.la
+libeedi3m_la_SOURCES += EEDI3/EEDI3CL.cpp \
+                        EEDI3/EEDI3CL.hpp \
+                        EEDI3/EEDI3CL_SSE2.cpp
 
 libclsse4_la_SOURCES = EEDI3/EEDI3CL_SSE4.cpp
 libclsse4_la_CXXFLAGS = $(AM_CXXFLAGS) -msse4.1
@@ -46,7 +48,7 @@ libclsse4_la_CXXFLAGS = $(AM_CXXFLAGS) -msse4.1
 libclavx2_la_SOURCES = EEDI3/EEDI3CL_AVX2.cpp
 libclavx2_la_CXXFLAGS = $(AM_CXXFLAGS) -mfma -mavx2
 
-libeedi3_la_LIBADD = libclsse4.la libclavx2.la
+libeedi3m_la_LIBADD += libclsse4.la libclavx2.la
 endif
 
-libeedi3_la_LDFLAGS = -no-undefined -avoid-version $(PLUGINLDFLAGS) $(OPENCLLDFLAGS)
+libeedi3m_la_LDFLAGS = -no-undefined -avoid-version $(PLUGINLDFLAGS) $(OPENCLLDFLAGS)
