From bd4d94f8c9eaf5257ebeecdf1f69808c5d80251b Mon Sep 17 00:00:00 2001
From: cantabile <cantabile.desu@gmail.com>
Date: Sun, 3 Apr 2016 16:06:29 +0300
Subject: [PATCH] Super: fix crash with chroma=False

Bug introduced in 691a7a99b7396f871ab3c6f9583becf890c6545b, no doubt.
---
 src/MVFrame.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/MVFrame.c b/src/MVFrame.c
index 3628e82..a9f8cec 100644
--- a/src/MVFrame.c
+++ b/src/MVFrame.c
@@ -618,14 +618,15 @@ void mvfDeinit(MVFrame *mvf) {
 
 void mvfUpdate(MVFrame *mvf, uint8_t **pSrc, int *pitch) {
     for (int i = 0; i < 3; i++) {
-        if (pSrc[i])
+        if (pSrc[i] && mvf->planes[i])
             mvpUpdate(mvf->planes[i], pSrc[i], pitch[i]);
     }
 }
 
 
 void mvfFillPlane(MVFrame *mvf, const uint8_t *pNewPlane, int nNewPitch, int plane) {
-    mvpFillPlane(mvf->planes[plane], pNewPlane, nNewPitch);
+    if (mvf->planes[plane])
+        mvpFillPlane(mvf->planes[plane], pNewPlane, nNewPitch);
 }
 
 
