diff --git a/FFT3DFilter.cpp b/FFT3DFilter.cpp
index c0bd179..2c9b18f 100644
--- a/FFT3DFilter.cpp
+++ b/FFT3DFilter.cpp
@@ -83,6 +83,7 @@
 
 #include <cstring>
 #include <algorithm>
+#include<cmath>
 
 #include "FFT3DFilter.h"
 #include "info.h"
diff --git a/fft3dfilter_c.cpp b/fft3dfilter_c.cpp
index 211a8fd..beb6bd6 100644
--- a/fft3dfilter_c.cpp
+++ b/fft3dfilter_c.cpp
@@ -22,6 +22,7 @@
  *****************************************************************************/
 
 #include <algorithm>
+#include<cmath>
 
 #include <fftw3.h>
 
