From d2c59bb4e4949a1b747d21f76494705c315b382a Mon Sep 17 00:00:00 2001
From: Austin Dworaczyk Wiltshire <561689+adworacz@users.noreply.github.com>
Date: Mon, 11 Aug 2025 23:34:42 -0700
Subject: [PATCH] Fix aarch64 builds by adjusting compilation flags (#4)

Fixes #3
---
 meson.build | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 23f1d25..cc39a31 100644
--- a/meson.build
+++ b/meson.build
@@ -43,11 +43,17 @@ libs += static_library('p2p_main',
     cpp_args: ['-std=c++20']
 )
 
+if host_machine.cpu_family().startswith('x86')
+  fpng_cpp_args = ['-std=c++20', '-DFPNG_NO_SSE=0', '-msse4.1', '-mpclmul', '-fno-strict-aliasing']
+else
+  fpng_cpp_args = ['-std=c++20', '-DFPNG_NO_SSE=1', '-fno-strict-aliasing']
+endif
+
 libs += static_library('fpng',
     [
         'src/fpng.cpp',
     ],
-    cpp_args: ['-std=c++20', '-msse4.1', '-mpclmul', '-fno-strict-aliasing']
+    cpp_args: fpng_cpp_args
 )
 
 shared_module('fpng', sources,
